pinot-core/src/test/java/org/apache/pinot/queries/RangeQueriesTest.java

```diff
@@ -97,7 +98,7 @@ protected List<IndexSegment> getIndexSegments() {
   private Set<String> _noDictionaryColumns;
   private Set<String> _rangeIndexColumns;
 
-  @BeforeClass
+  @BeforeMethod
```

Comment:
We don't want to construct a new segment per method. We can share the same segment across all tests.
You may add an `@AfterClass` method (we usually call it `tearDown()`) to destroy the segment and delete the dir. Closing `_indexSegment` is enough as `_indexSegments` is just a reference to it